src/tests.rs

```diff
@@ -43,6 +45,60 @@ mod tests {
         assert_eq!(retval, Some(6));
     }
 
+    #[pg_test]
+    #[search_path(@extschema@)]
+    fn plrust_composite_type() {
+        let definition = r#"
+            CREATE TYPE Dog AS (
+                name TEXT,
+                scritches INT
+            );
+
+            CREATE FUNCTION last_dog(a Dog[]) RETURNS Dog
+                IMMUTABLE STRICT
+                LANGUAGE PLRUST AS
+            $$
+                a.into_iter().last().unwrap_or(None)
+            $$;
+        "#;
+        Spi::run(definition);
+
+        // Since it's a composite type, we need to look it up
+        let composite_type_oid: u32 = unsafe {
+            let mut typoid = 0;
+            let mut typmod = 0;
+            pg_sys::parseTypeString("Dog".as_pg_cstr(), &mut typoid, &mut typmod, false);
+            typoid
+        };
```

Comment:
Need to find a way to wrap this outside this function before this is acceptable.